jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialHelper.java

```diff
@@ -73,19 +76,38 @@
    * @param credentialHelper the path to the credential helper executable
    */
   public DockerCredentialHelper(String serverUrl, Path credentialHelper) {
-    this(serverUrl, credentialHelper, System.getProperties(), ProcessBuilder::new);
+    this(
+        serverUrl,
+        credentialHelper,
+        System.getProperties(),
+        ProcessBuilder::new,
+        Collections.emptyMap());
+  }
+
+  /**
+   * Constructs a new {@link DockerCredentialHelper}.
+   *
+   * @param serverUrl the server URL to pass into the credential helper
+   * @param credentialHelper the path to the credential helper executable
+   * @param environment environment variables used in configuring the credential helper
+   */
+  public DockerCredentialHelper(
```

Comment:
Not strictly - I added it to avoid refactoring code where the previous constructor was used, specifically the places where we didn't have access to any environment variables that could be used to configure the cred helper.

Doesn't seem to be a lot of places where the previous constructor was used though - can refactor as needed.

Reply:
Ah, I see. Yes the `DockerCredentialHelper(String serverUrl, Path credentialHelper, Map<String, String> environment)` constructor was being used in non-test classes. It's the constructor being used in `CredentialRetrieverFactory forImage(ImageReference imageReference, Consumer<LogEvent> logger, Map<String, String> environment)`. Np - I'll add it back in